File: t/udp-generator.py
#!/usr/bin/python3

import sys
import socket

if len(sys.argv) < 4:
	print('Usage: %s [server] [port] [input...]' % sys.argv[0])
	sys.exit(1)

server = sys.argv[1]
port = int(sys.argv[2])

s = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
s.connect((server, port))

for f in sys.argv[3:]:
	file = open(f, mode='rb')
	msg = file.read()
	s.send(msg)

File: t/udp-generator.pl
#!/usr/bin/perl

# Read datagrams from input files and send them to a specified server

use strict;
use warnings;
use IO::File;
use IO::Socket;

if (@ARGV < 3) {
	print STDERR "Usage: $0 [server] [port] [input...]\n";
	exit 1;
}

my $server = shift @ARGV;
my $port = shift @ARGV;

my $sock = IO::Socket::INET->new(
	Type     => SOCK_DGRAM,
	Proto    => 'udp',
	PeerAddr => $server,
	PeerPort => $port,
) or die "Failed to create a UDP socket: $!\n";

for my $f (@ARGV) {
	my $fh = IO::File->new($f, "r") or die "Failed to open file: $f: $!\n";
	my $read = $fh->read(my $dgram, 1500); # 1500 bytes must be enough to store the entire datagram at the moment
	undef $fh;
	my $sent = $sock->send($dgram) or die "Failed to send a datagram: $!\n";
	if ($read != $sent) {
		die "$read bytes read but $sent bytes sent\n";
	}
}

$sock->close();
